src/core/net/ip6_address.cpp

```diff
@@ -45,6 +45,30 @@ using ot::Encoding::BigEndian::HostSwap32;
 namespace ot {
 namespace Ip6 {
 
+bool InterfaceIdentifier::operator==(const InterfaceIdentifier &aOther) const
+{
+    return memcmp(m8, aOther.m8, sizeof(InterfaceIdentifier)) == 0;
+}
+
+bool InterfaceIdentifier::IsUnspecified(void) const
+{
+    return (m8[0] == 0 && m8[1] == 0 && m8[2] == 0 && m8[3] == 0 && m8[4] == 0 && m8[5] == 0 && m8[6] == 0 &&
+            m8[7] == 0);
+}
+
+bool InterfaceIdentifier::IsReserved(void) const
+{
+    Address addr;
+
+    addr.SetIid(this->m8);
+    return addr.IsIidReserved();
```

Comment:
Agree it's also an option.
However there are already a set of `Iid` relative APIs in `class Address`, not just the Reserved ones.

https://github.com/openthread/openthread/blob/752ad977db187ab9357621ffd7e6247ef075f481/src/core/net/ip6_address.hpp#L293-L342



The choice of existing option is to avoid breaking the previous elegance due to this new interface to check whether or not it is reserved.

Reply:
Any preference? @jwhui